documentation/src/main/asciidoc/topics/con_expiration_overview.adoc

```diff
@@ -0,0 +1,38 @@
+[id='how_expiration_works-{context}']
+= How Expiration Works
+By default, `Cache.put()` creates immortal entries. With expiration, `Cache.put()` creates mortal entries with lifespans and maximum idle times.
+
+When you configure expiration, {brandname} stores keys with metadata that
+determines when entries expire.
+
+* Lifespan uses a `creation` timestamp and the value for the `lifespan` configuration property.
+* Maximum idle uses a `last used` timestamp and the value for the `max-idle` configuration property.
+
+{brandname} checks if lifespan or maximum idle metadata is set and then
+compares the values with the current time.
+
+If `(creation + lifespan > currentTime)` or `(lastUsed + maxIdle > currentTime)` then {brandname} detects that the entry is expired.
+
+Expiration occurs whenever entries are accessed or iterated on.
+
+For example, `k1` reaches the maximum idle time and a client makes a
+`Cache.get(k1)` request. In this case, {brandname} detects that the entry is
+expired and removes it. The `Cache.get()` call does not return.
+
+//dnaro: contradicting info here. does this apply with local cache stores only?
```

Comment:
So technically this is a bit different based on if it is a store vs a loader. Technically a loader would still load the value if it expired. The vast majority of cases are a store though.

Reply:
@wburns does this mean expiration doesn't work with cache loaders? should we mention that?

Reply:
No, it does work just fine.  You have to think of a loader vs a store differently. A Loader can only load values into memory from an external location. A store can update this external location. Thus an expiration cannot write to a Loader, however it can remove the in memory contents. We have use cases where a user wants to load from a Loader and have expiration so the value is only so stale for example.